be/src/exprs/jsonpath.cpp

```diff
@@ -52,6 +56,296 @@ bool ArraySelectorSlice::match(const std::string& input) {
     return RE2::FullMatch(input, ARRAY_SLICE_SELECTOR);
 }
 
+static inline std::string trim_copy(const std::string& s) {
+    size_t i = 0;
+    while (i < s.size() && std::isspace(static_cast<unsigned char>(s[i]))) i++;
+    size_t j = s.size();
+    while (j > i && std::isspace(static_cast<unsigned char>(s[j - 1]))) j--;
+    return s.substr(i, j - i);
+}
+
+bool ArraySelectorFilter::match(const std::string& input) {
+    return RE2::FullMatch(trim_copy(input), ARRAY_FILTER_SELECTOR);
+}
+
+// Minimal evaluator for filter expressions: ?(@.field op literal) with && and ||, parentheses.
+// Supported ops: =, !=, >, <, >=, <=; literals: numbers and double/single-quoted strings.
+namespace {
+
+enum class TokenType {
+    END,
+    LPAREN,
+    RPAREN,
+    AND,
+    OR,
+    OP_EQ,
+    OP_NE,
+    OP_GT,
+    OP_LT,
+    OP_GE,
+    OP_LE,
+    AT,
+    DOT,
+    IDENT,
+    NUMBER,
+    STRING
+};
+
+struct Token {
+    TokenType type;
+    std::string text;
+};
+
+struct Lexer {
+    const std::string& s;
+    size_t i = 0;
+    explicit Lexer(const std::string& str) : s(str) {}
+    static bool is_ident_start(char c) { return std::isalpha(static_cast<unsigned char>(c)) || c == '_' || c == '$'; }
+    static bool is_ident_char(char c) { return std::isalnum(static_cast<unsigned char>(c)) || c == '_' || c == '$'; }
+    void skip_ws() {
+        while (i < s.size() && std::isspace(static_cast<unsigned char>(s[i]))) i++;
+    }
+    Token next() {
+        skip_ws();
+        if (i >= s.size()) return {TokenType::END, {}};
+        char c = s[i];
+        if (c == '(') {
```

Comment:
looks like these if-check can be optimized with a map lookup, like `{'(' -> TokenType::LPAREN, ')' -> TokenType::RPAREN}`